src/Asv.Avalonia.Example/Pages/FileBrowser/FileBrowserViewModel.cs

```diff
@@ -0,0 +1,814 @@
+                var relativeFile = Path.GetRelativePath(_localRootPath, file);
+                var id = NavigationId.NormalizeTypeId(relativeFile);
+                var parentPath = Directory.GetParent(file)?.FullName ?? _localRootPath;
+                var fileInfo = new FileInfo(file);
+
+                lock (items)
+                {
+                    items.Add(new FileItem(id, parentPath, file, fileInfo.Name, fileInfo.Length));
+                }
+            }
+        );
+    }
+
+    private ObservableList<IBrowserItem> LoadRemoteItems()
+    {
+        var items = new ObservableList<IBrowserItem>();
+
+        ClientEx.Entries.ForEach(e =>
+        {
+            if (e.Value.Path == MavlinkFtpHelper.DirectorySeparator.ToString())
+            {
+                var root = new DirectoryItem(
+                    "_",
+                    string.Empty,
+                    MavlinkFtpHelper.DirectorySeparator.ToString(),
+                    "_"
+                );
+                items.Add(root);
+                return;
+            }
+
+            var item = e.Value.Type switch
+            {
+                FtpEntryType.Directory => new DirectoryItem(
+                    NavigationId.NormalizeTypeId(e.Value.Path),
+                    e.Value.ParentPath,
+                    e.Key,
+                    e.Value.Name
+                ),
+                FtpEntryType.File => new FileItem(
+                    NavigationId.NormalizeTypeId(e.Value.Path),
+                    e.Value.ParentPath,
+                    e.Key,
+                    e.Value.Name,
+                    ((FtpFile)e.Value).Size
+                ),
+                _ => new BrowserItem(
+                    NavigationId.NormalizeTypeId(e.Value.Path),
+                    e.Value.ParentPath,
+                    e.Key
+                ),
+            };
+
+            items.Add(item);
+        });
+
+        return items;
+    }
+
+    #endregion
+
+    private static string Crc32ToHex(uint crc32) => crc32.ToString("X8");
+
+    public override ValueTask<IRoutable> Navigate(NavigationId id)
+    {
+        return ValueTask.FromResult<IRoutable>(this);
+    }
+
+    public override IEnumerable<IRoutable> GetRoutableChildren()
+    {
+        return [];
+    }
+
+    protected override void AfterLoadExtensions() { }
+
+    protected override void AfterDeviceInitialized(IClientDevice device)
+    {
+        Title.Value = $"Browser[{device.Id}]";
+        Client = device.GetMicroservice<IFtpClient>();
+        ArgumentNullException.ThrowIfNull(Client);
+        ClientEx = device.GetMicroservice<IFtpClientEx>() ?? new FtpClientEx(Client);
+        InitCommands();
+    }
+
+    public override IExportInfo Source => SystemModule.Instance;
+
+    #region Dispose
+
+    private IDisposable _sub1;
+
+    protected override void Dispose(bool disposing)
+    {
+        if (disposing)
+        {
+            _sub1.Dispose();
+
+            _localSource.Clear();
```

Comment:
нельзя делать обычный клиар, так как он не запускает диспоуз. Только ремув, а так же обязательно подписываться на дипсоуз при ремуве у коллекций, чтобы все точно диспоузилось
```suggestion
_remoteSource.RemoveRange(0, _remoteSource.Count);
```